tests/suites/test_suite_mpi.function

```diff
@@ -763,6 +763,316 @@ exit:
+
+    ASSERT_ALLOC( X, len );
+    ASSERT_ALLOC( Y, len );
+
+    TEST_ASSERT( mbedtls_mpi_core_read_be( X, limbs, input_X->x, input_X->len )
+                 == 0 );
+
+    TEST_ASSERT( mbedtls_mpi_core_read_be( Y, limbs, input_Y->x, input_Y->len )
+                 == 0 );
+
+    /* condition is false */
+    TEST_CF_SECRET( X, len );
+    TEST_CF_SECRET( Y, len );
+
+    mbedtls_mpi_core_cond_assign( X, Y, copy_limbs, 0 );
+
+    TEST_CF_PUBLIC( X, len );
+    TEST_CF_PUBLIC( Y, len );
+
+    TEST_ASSERT( memcmp( X, Y, len ) != 0 );
+
+    /* condition is true */
+    TEST_CF_SECRET( X, len );
+    TEST_CF_SECRET( Y, len );
+
+    mbedtls_mpi_core_cond_assign( X, Y, copy_limbs, 1 );
+
+    TEST_CF_PUBLIC( X, len );
+    TEST_CF_PUBLIC( Y, len );
+
+    /* Check if the given length is copied even it is smaller
+       than the length of the given MPIs. */
+    if( copy_limbs != limbs )
+    {
+        ASSERT_COMPARE( X, copy_len, Y, copy_len );
+        TEST_ASSERT( memcmp( X, Y, len ) != 0 );
+    }
+    else
+        ASSERT_COMPARE( X, len, Y, len );
+
+exit:
+    mbedtls_free( X );
+    mbedtls_free( Y );
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void mpi_core_cond_swap( data_t * input_X,
+                         data_t * input_Y,
+                         int input_len )
+{
+    mbedtls_mpi_uint *tmp_X = NULL;
+    mbedtls_mpi_uint *tmp_Y = NULL;
+    mbedtls_mpi_uint *X = NULL;
+    mbedtls_mpi_uint *Y = NULL;
+    size_t limbs_X = CHARS_TO_LIMBS( input_X->len );
+    size_t limbs_Y = CHARS_TO_LIMBS( input_Y->len );
+    size_t limbs = limbs_X;
+    size_t copy_limbs = CHARS_TO_LIMBS( input_len );
+    size_t len = limbs * sizeof( mbedtls_mpi_uint );
+    size_t copy_len = copy_limbs * sizeof( mbedtls_mpi_uint );
+
+    TEST_EQUAL( limbs_X, limbs_Y );
+    TEST_ASSERT( copy_limbs <= limbs );
+
+    ASSERT_ALLOC( tmp_X, len );
+    ASSERT_ALLOC( tmp_Y, len );
+
+    TEST_ASSERT( mbedtls_mpi_core_read_be( tmp_X, limbs,
+                                           input_X->x, input_X->len )
+                 == 0 );
+    ASSERT_ALLOC( X, len );
+    memcpy( X, tmp_X, len );
+
+    TEST_ASSERT( mbedtls_mpi_core_read_be( tmp_Y, limbs,
+                                           input_Y->x, input_Y->len )
+                 == 0 );
+    ASSERT_ALLOC( Y, len );
+    memcpy( Y, tmp_Y, len );
+
+    /* condition is false */
+    TEST_CF_SECRET( X, len );
+    TEST_CF_SECRET( Y, len );
+
+    mbedtls_mpi_core_cond_swap( X, Y, copy_limbs, 0 );
+
+    TEST_CF_PUBLIC( X, len );
+    TEST_CF_PUBLIC( Y, len );
+
+    ASSERT_COMPARE( X, len, tmp_X, len );
+    ASSERT_COMPARE( Y, len, tmp_Y, len );
+
+    /* condition is true */
+    TEST_CF_SECRET( X, len );
+    TEST_CF_SECRET( Y, len );
+
+    mbedtls_mpi_core_cond_swap( X, Y, copy_limbs, 1 );
+
+    TEST_CF_PUBLIC( X, len );
+    TEST_CF_PUBLIC( Y, len );
+
+    /* Check if the given length is copied even it is smaller
+       than the length of the given MPIs. */
+    if( copy_limbs != limbs )
```

Comment:
Again, if we're testing that the code isn't writing past the end of the destination, should we also check it doesn't write before the beginning?